src/server/server.c

```diff
@@ -121,6 +121,12 @@ queue_message(TSpeechDMessage * new, int fd, int history_flag,
 		new->time = time(NULL);
 
 		new->settings.paused_while_speaking = 0;
+
+		/* Ignore possible SSML mode for anything but TEXT, as it doesn't make
+		 * sense anything else and can cause problems if the message processing
+		 * assumes SSML although it isn't */
+		if (type != SPD_MSGTYPE_TEXT)
+			new->settings.ssml_mode = SPD_DATA_TEXT;
```

Comment:
Why inside the `if (fd>0)` ? a per-uid ssml parameterization would seem odd, but better be safe than sorry?

Reply:
Just because the comment above the function says that *[…] the message _new_ already contains a fully filled in settings structure which should not be overwritten […]*.

I assumed that the message then might already have been partly processed, or that the settings member could be shared between messages so I couldn't restore the SSML flag for TEXT messages needing it.  But all this was merely an assumption seeing all settings-related stuff that was not directly a argument to the function was done there, I didn't actually dig deep enough to understand in which cases it's called with an uid and the detailed implications.
I can change this if needed.

Reply:
Ok, I see, the caller with nul fd indeed is just re-queueing the message, so it was already processed and fixed, thus no need to fix it again.

Thanks!